include/fuse_lowlevel.h

```diff
@@ -1845,6 +1845,10 @@ int fuse_lowlevel_notify_delete(struct fuse_session *se,
  * If the stored data overflows the current file size, then the size
  * is extended, similarly to a write(2) on the filesystem.
  *
+ * To avoid a deadlock this function must not be called while executing
+ * a related filesystem operation (e.g. while replying to a FUSE_READ
+ * request).
+ *
```

Comment:
I think all notification should not be called while executing fuse kernel requests. I wonder if we should better create a file like docs/README-notifications.1st?

Reply:
OK, that probably makes more sense.  Let me try to write something to share through this MR.  Thanks @bsbernd